ios/Helpers.swift

```diff
@@ -247,3 +255,140 @@ func parseWorkoutConfiguration(_ dict: NSDictionary) -> HKWorkoutConfiguration {
 
     return configuration
 }
+
+#if compiler(>=6)
+@available(iOS 18.0, *)
+extension HKStateOfMind.Kind: @retroactive CaseIterable, @retroactive CustomStringConvertible {
+  public var description: String {
+    switch self {
+      case .dailyMood: "Daily mood"
+      case .momentaryEmotion: "Momentary Emotion"
+      @unknown default: "Unknown"
+    }
+  }
+
+  public static var allCases: [HKStateOfMind.Kind] {
+    [.dailyMood, .momentaryEmotion]
+  }
+
+  static func convertToStateOfMindKind(int: Int) -> HKStateOfMind.Kind {
+    // default to .dailyMood if we receive an int out of bounds
+    return HKStateOfMind.Kind(rawValue: int) ?? .dailyMood
+  }
+}
+#endif
+
+#if compiler(>=6)
+@available(iOS 18.0, *)
+extension HKStateOfMind.Label: @retroactive CaseIterable, @retroactive CustomStringConvertible {
+  public var description: String {
+    switch self {
```

Comment:
I’m not too familiar with Swift, but are these switches necessary? It looks like you can convert an enumeration to a string using kind.rawValue, and in this case, it might just be a matter of uppercasing the first letter.

Let me know if I’m missing something or if this makes sense!